pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -66,6 +66,9 @@
   private TimeFieldSpec _timeFieldSpec;
   private final List<DateTimeFieldSpec> _dateTimeFieldSpecs = new ArrayList<>();
   private final List<ComplexFieldSpec> _complexFieldSpecs = new ArrayList<>();
+  // names of the columns that used as primary keys
+  // TODO: add validation checks like duplicate columns and use of time column
+  private List<String> _primaryKeyColumns;
```

Comment:
This should not be part of the schema, but the upsert config

Reply:
Primary key is an inherent attribute of the schema. And its usage is beyond the upsert. For example, it's also used in the join feature which is happening in parallel.